### PR TITLE
[WebProfilerBundle] Allow inject WebDebugToolbar to StreamedResponse

### PR DESCRIPTION
**Pull request by bkrukowski**
I would like to have an option to inject `WebDebugToolbar` to `StreamedResponse`.

| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

**Comment by fabpot** [comment]
fabbot reports errors that should be fixed.

**Comment by nicolas-grekas** [comment]
Related to #20391 I guess

**Comment by stof** [comment]
I don't see how this solve the issue for StreamedResponse, as it does not implement the interface

File: src/Symfony/Bundle/WebProfilerBundle/Tests/EventListener/TestStreamedResponseHtml.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sendContent()
+    {
+        ob_start();
+        parent::sendContent();
+        $content = ob_get_contents();
+        ob_end_clean();
```
Comment (stof) [inline src/Symfony/Bundle/WebProfilerBundle/Tests/EventListener/TestStreamedResponseHtml.php]: doing an implementation based on a bugger totally defeats the purpose of using a streamed response, as it is not streamed anymore. So this does not show that you can solve your issue (using your TestStreamedResponseHtml ends up being equivalent to using a normal Response regarding the sending of the response)

Comment (bkrukowski) [inline src/Symfony/Bundle/WebProfilerBundle/Tests/EventListener/TestStreamedResponseHtml.php]: @stof this class is only for tests

**Comment by bkrukowski** [comment]
@fabpot Yes, it is related, but my case is little bit different. For me does not matter if debugger will be placed before `<title>` or `</body>`. Currently I have the following code:

```php
    public function onKernelException(GetResponseForExceptionEvent $event)
    {
        $response = new Response();
        // ...
        ob_start();
        $view->display();
        $content = ob_get_contents();
        ob_end_clean();
        $response->setContent($content);

        $event->setResponse($response);
    }
```
and I would like to rewrite code to
```php
    public function onKernelException(GetResponseForExceptionEvent $event)
    {
        $response = new StreamedResponse();
        $myHtmlStreamedResponse = new MyHtmlStreamedResponse();
        $myHtmlStreamedResponse->setCallback(function () use ($myHtmlStreamedResponse) {
            // ...
            $view->appendBody($myHtmlStreamedResponse->getExtraHtml());
            $view->display();
        });
        $event->setResponse($response);
    }
```
and in template:
```twig
{# ... #}
{{ extraHtml }}
</body>
</html>
```

@stof please take a look on my example. In this case whole HTML will be not loaded to memory.

File: src/Symfony/Bundle/WebProfilerBundle/Tests/EventListener/TestStreamedResponseHtml.php (hunk 1 above)
Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/Tests/EventListener/TestStreamedResponseHtml.php]: what @stof means is that the test should reflect what you are trying to implement, so it must be a streamed response.

Comment (bkrukowski) [inline src/Symfony/Bundle/WebProfilerBundle/Tests/EventListener/TestStreamedResponseHtml.php]: @stof please check now

**Comment by javiereguiluz** [comment]
@bkrukowski thanks (and congrats) for your first contribution to Symfony!

However, this proposal looks a bit complex to me. And the provided example, using a StreamedResponse in an Exception listener looks confusing to me.

This is an unpopular opinion, but I don't find StreamedResponse that useful now that we can do everything in browsers with websockets + JavaScript.

**Comment by nicolas-grekas** [comment]
Trying to help move forward, I'm :-1: here. Instead, just create your own listener that extends WebDebugToolbarListener and replaces the injectToolbar() method by one of yours.
Closing as previous comments were raising concerns also.

**Pull request closed without merging**